dhis-2/dhis-web/dhis-web-commons/src/main/java/org/hisp/dhis/servlet/filter/AppOverrideFilter.java

```diff
@@ -0,0 +1,175 @@
+
+    // -------------------------------------------------------------------------
+    // Filter implementation
+    // -------------------------------------------------------------------------
+
+    @Override
+    public void init( FilterConfig config )
+    {
+    }
+
+    // From AppController.java (some duplication)
+    private void serveInstalledAppResource( App app, String resourcePath, HttpServletRequest request, HttpServletResponse response )
+        throws IOException {
+        // Get page requested
+
+        log.debug( String.format( "Serving app resource: '%s'", resourcePath ) );
+
+        // Handling of 'manifest.webapp'
+        if ( "manifest.webapp".equals( resourcePath ) )
+        {
+            // If request was for manifest.webapp, check for * and replace with host
+            if ( "*".equals( app.getActivities().getDhis().getHref() ) )
+            {
+                String contextPath = "../";
+                log.debug( String.format( "Manifest context path: '%s'", contextPath ) );
+                app.getActivities().getDhis().setHref( contextPath );
+            }
+
+            jsonMapper.writeValue( response.getOutputStream(), app );
+        }
+        // Any other resource
+        else
+        {
+            // Retrieve file
+            Resource resource = appManager.getAppResource( app, resourcePath );
+
+            if ( resource == null )
+            {
+                response.sendError( HttpServletResponse.SC_NOT_FOUND );
+                return;
+            }
+
+            String filename = resource.getFilename();
+            log.debug( String.format( "App filename: '%s'", filename ) );
+
+            if ( new ServletWebRequest( request, response ).checkNotModified( resource.lastModified() ) )
+            {
+                response.setStatus( HttpServletResponse.SC_NOT_MODIFIED );
+                return;
+            }
+
+            String mimeType = request.getSession().getServletContext().getMimeType( filename );
+
+            if ( mimeType != null )
+            {
+                response.setContentType( mimeType );
+            }
+
+            response.setContentLength( (int) resource.contentLength() );
+            response.setHeader( "Last-Modified", DateUtils.getHttpDateString( new Date( resource.lastModified() ) ) );
+            StreamUtils.copy( resource.getInputStream(), response.getOutputStream() );
+        }
+    }
+
+    @Override
+    public void doFilter( ServletRequest req, ServletResponse res, FilterChain chain )
+        throws IOException, ServletException {
+        HttpServletRequest request = (HttpServletRequest) req;
+        HttpServletResponse response = (HttpServletResponse) res;
+        String requestURI = request.getRequestURI();
+
+        Pattern p = Pattern.compile( APP_PATH_PATTERN );
+        Matcher m = p.matcher( requestURI );
+
+        if ( m.find() ) {
+            String namespace = m.group( 0 );
```

Comment:
Apply DHIS 2 code style.

Reply:
Where is the code style documented?  Is there a lint tool?  @stian-sandvold [made the code-style recommendation](https://github.com/dhis2/dhis2-core/pull/5977#discussion_r475529098) (spaces around the 0) on this line

Reply:
You can find info here: https://github.com/dhis2/wow-backend/blob/master/guides/code_formatting.md

Reply:
Here specifically, brackets go on new lines.

Reply:
Applied codeformatting